x-pack/plugins/fleet/public/components/manage_agent_policies_modal.tsx

```diff
@@ -0,0 +1,213 @@
+  );
+
+  const onCancel = () => {
+    onClose();
+  };
+
+  const onConfirm = async () => {
+    setIsSubmitting(true);
+    const { error } = await savePackagePolicy({
+      policy_ids: selectedPolicyIds,
+    });
+    setIsSubmitting(false);
+    if (!error) {
+      onAgentPoliciesChange();
+      notifications.toasts.addSuccess({
+        title: i18n.translate('xpack.fleet.manageAgentPolicies.updatedNotificationTitle', {
+          defaultMessage: `Successfully updated ''{packagePolicyName}''`,
+          values: {
+            packagePolicyName: packagePolicy.name,
+          },
+        }),
+        'data-test-subj': 'policyUpdateSuccessToast',
+      });
+    } else {
+      if (error.statusCode === 409) {
+        notifications.toasts.addError(error, {
+          title: i18n.translate('xpack.fleet.manageAgentPolicies.failedNotificationTitle', {
+            defaultMessage: `Error updating ''{packagePolicyName}''`,
+            values: {
+              packagePolicyName: packagePolicy.name,
+            },
+          }),
+          toastMessage: i18n.translate(
+            'xpack.fleet.manageAgentPolicies.failedConflictNotificationMessage',
+            {
+              defaultMessage: `Data is out of date. Refresh the page to get the latest policy.`,
+            }
+          ),
+        });
+      } else {
+        notifications.toasts.addError(error, {
+          title: i18n.translate('xpack.fleet.manageAgentPolicies.failedNotificationTitle', {
+            defaultMessage: `Error updating ''{packagePolicyName}''`,
+            values: {
+              packagePolicyName: packagePolicy.name,
+            },
+          }),
+        });
+      }
+    }
+    onClose();
+  };
+
+  const { agentPolicyMultiOptions, isLoading } = useAgentPoliciesOptions(packageInfo);
+
+  return (
+    <EuiConfirmModal
+      title={
+        <FormattedMessage
+          id="xpack.fleet.manageAgentPolicies.confirmModalTitle"
+          defaultMessage="Manage agent policies"
+        />
+      }
+      onCancel={onCancel}
+      onConfirm={onConfirm}
+      cancelButtonText={
+        <FormattedMessage
+          id="xpack.fleet.manageAgentPolicies.confirmModalCancelButtonLabel"
+          defaultMessage="Cancel"
+        />
+      }
+      confirmButtonText={
+        <FormattedMessage
+          id="xpack.fleet.manageAgentPolicies.confirmModalConfirmButtonLabel"
+          defaultMessage="Confirm"
+        />
+      }
+      buttonColor="primary"
+      confirmButtonDisabled={
+        selectedPolicyIds.length === 0 ||
+        isSubmitting ||
+        isEqual(initialPolicyIds, selectedPolicyIds)
+      }
+      data-test-subj="manageAgentPoliciesModal"
+    >
+      <EuiFlexGroup direction="column" gutterSize="m">
+        <EuiFlexItem>
+          <EuiText>
+            <FormattedMessage
+              id="xpack.fleet.manageAgentPolicies.confirmModalDescription"
+              defaultMessage="Agent policies which share this integration"
```

Comment:
maybe it would sound better `Agent policies sharing this integration`?

Reply:
good idea, updated